inference-engine/src/mkldnn_plugin/nodes/psroi.cpp

```diff
@@ -46,24 +53,29 @@ class PSROIPoolingImpl: public ExtLayerBase {
             part_size_ = layer->GetParamAsInt("part_size", 1);
             trans_std_ = layer->GetParamAsFloat("trans_std", 1);
 
+            bool isBf16Input = layer->insData[0].lock()->getTensorDesc().getPrecision() == Precision::BF16;
             if (no_trans_) {
-                addConfig(layer, {DataConfigurator(ConfLayout::PLN, Precision::FP32), DataConfigurator(ConfLayout::PLN, Precision::FP32)},
-                    {DataConfigurator(ConfLayout::PLN, Precision::FP32)});
+                addConfig(layer, {DataConfigurator(ConfLayout::PLN, isBf16Input ? Precision::BF16 : Precision::FP32), DataConfigurator(ConfLayout::PLN)},
+                          {DataConfigurator(ConfLayout::PLN, isBf16Input ? Precision::BF16 : Precision::FP32)});
             } else {
-                addConfig(layer, {DataConfigurator(ConfLayout::PLN, Precision::FP32), DataConfigurator(ConfLayout::PLN, Precision::FP32),
-                                  DataConfigurator(ConfLayout::PLN, Precision::FP32)}, {DataConfigurator(ConfLayout::PLN, Precision::FP32)});
+                addConfig(layer, {DataConfigurator(ConfLayout::PLN, isBf16Input ? Precision::BF16 : Precision::FP32), DataConfigurator(ConfLayout::PLN),
+                                  DataConfigurator(ConfLayout::PLN)}, {DataConfigurator(ConfLayout::PLN, isBf16Input ? Precision::BF16 : Precision::FP32)});
             }
         } catch (InferenceEngine::details::InferenceEngineException &ex) {
             errorMsg = ex.what();
         }
     }
 
-    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs,
-                       ResponseDesc *resp) noexcept override {
-        float* dst_data = outputs[0]->buffer();
-        const float *bottom_data_beginning = inputs[0]->buffer();
+    template <typename inputType, typename outputType>
+    StatusCode executeSpecified(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs,
+                                ResponseDesc *resp) {
+        float* dst_data_fp32 = outputs[0]->buffer();
+        const float *bottom_data_fp32 = inputs[0]->buffer();
         const float *bottom_rois_beginning = inputs[1]->buffer();
 
+        auto src_data = reinterpret_cast<const inputType *>(bottom_data_fp32);
+        auto dst_data = reinterpret_cast<outputType *>(dst_data_fp32);
+
```

Comment:
It might be better to use a type cast request from memory blob interface like the following:
```c++
        const auto *src_data = inputs[0]->cbuffer().as<const inputType*>();
        auto *dst_data = outputs[0]->buffer().as<outputType*>();
```
Or for example inline the `dst_data_fp32` and `bottom_data_fp32` variables.

Reply:
done